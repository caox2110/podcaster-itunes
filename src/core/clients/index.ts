export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
