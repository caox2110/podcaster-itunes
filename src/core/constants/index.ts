import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
