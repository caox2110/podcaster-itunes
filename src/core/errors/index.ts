export { default as ExceptionHandler } from './Exceptions.handler';
