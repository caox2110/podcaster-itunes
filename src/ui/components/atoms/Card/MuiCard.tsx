import Card from '@mui/material/Card';
import type { CardProps } from '@mui/material/Card';

/**
 * Card Component definition.
 *
 * @export
 * @param {CardProps} { ...rest }
 * @returns {*}
 */
export default function MuiCard({ ...rest }: CardProps) {
  return <Card {...rest} />;
}
