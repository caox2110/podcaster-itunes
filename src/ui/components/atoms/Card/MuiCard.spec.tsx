import type { CardProps } from '@mui/material/Card';

import { render } from '@/core/test';
import type { RenderResult } from '@/core/test';

import MuiCard from './MuiCard';

const properties = {};

function renderApp({ ...rest }: CardProps = properties): RenderResult {
  const utils = render(<MuiCard {...rest} />, { withRouter: true });

  return {
    ...utils,
  };
}

describe('MuiCard', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
