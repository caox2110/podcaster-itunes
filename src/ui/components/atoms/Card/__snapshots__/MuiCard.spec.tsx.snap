// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiCard Rendering Should render without throwing any errors 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;
